client/www/lib/feedback/RatingBox.tsx

```diff
@@ -0,0 +1,165 @@
+        <div className="w-20">
+          <ToggleGroup
+            selectedId={selectedId}
+            onChange={(item) => {
+              if (!previousRating) {
+                setShouldAutoFocus(true);
+              }
+              clientDB.transact(
+                clientDB.tx.ratings[previousRating?.id ?? id()]
+                  .ruleParams({ localId })
+                  .update({
+                    pageId,
+                    localId,
+                    key: `${localId}_${pageId}`,
+                    wasHelpful: 'yes' === item.id,
+                  }),
+              );
+            }}
+            items={[
+              { id: 'yes', label: 'Yes' },
+              { id: 'no', label: 'No' },
+            ]}
+          />
+        </div>
+      </div>
+      {previousRating && selectedId && (
+        <div className="space-y-2">
+          <p>Thank you for your feedback! More details to share?</p>
+          <SavingTextArea
+            savedValue={previousRating.extraComment || ''}
+            onSave={(extraComment) => {
+              clientDB.transact(
+                clientDB.tx.ratings[previousRating.id]
+                  .ruleParams({ localId })
+                  .update({ extraComment }),
+              );
+            }}
+            placeholder="Tell us more about your experience..."
+            autoFocus={shouldAutoFocus}
+          />
+        </div>
+      )}
+    </div>
+  );
+}
+
+// ----------
+// Components
+
+type SavingTextAreaProps = {
+  savedValue: string;
+  onSave: (value: string) => void;
+} & Omit<
+  React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+  'value' | 'onChange' | 'onKeyDown'
+>;
+
+/**
+ * A handy textarea that lets you save a value.
+ * If the incoming `savedValue` changes, we'll update the textarea.
+ * We'll _skip_ the update if you are focused on the input
+ * and in the middle of making a change though! :)
+ */
+function SavingTextArea({ savedValue, onSave, ...props }: SavingTextAreaProps) {
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [value, setValue] = useState(savedValue);
+  useEffect(() => {
+    const ref = textAreaRef.current!;
+    const isEditing = ref === document.activeElement;
+    if (isEditing) return;
+    setValue(savedValue);
+  }, [savedValue]);
+  return (
+    <div className="space-y-1">
+      <textarea
+        {...props}
+        value={value}
+        className="flex w-full flex-1 rounded-sm border-gray-200 bg-white px-3 py-1 placeholder:text-gray-400 disabled:text-gray-400"
+        onChange={(e) => {
+          setValue(e.target.value);
+        }}
+        onKeyDown={(e) => {
+          if (e.key === 'Enter' && e.metaKey) {
+            onSave(value);
+          }
+        }}
+      />
+      <div className="text-right">
+        <Button
+          disabled={value === savedValue}
+          onClick={() => {
+            onSave(value);
+          }}
+          size="mini"
+          type="submit"
+        >
+          Save
```

Comment:
Maybe we could make this say `Saved!` if `value && value === savedValue`?

When I clicked "Save", it felt like it didn't do enough and I wasn't 100% sure it actually saved.

An alternative would be to let the user add many comments on the page and when you click "submit" it clears the input and shows the comment at the bottom of the page. Then they can leave multiple comments on the same page.

Reply:
Ah, good idea! Will start with `value && value === savedValue` 🫡